### PR TITLE
Add config to control if zorder using global sort

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Add a new config `spark.sql.optimizer.zorderGlobalSort.enabled` to control if we do a global sort using zorder. This is a trade-off with data skew if the zorder columns has low cardinality.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @cfmcgrady @yaooqinn @pan3793

**Comment by pan3793** [comment]
Seems the document is out-of-date, could you please also update doc?

**Review by yaooqinn — APPROVED** [review]

File: dev/kyuubi-extension-spark-common/src/test/scala/org/apache/spark/sql/ZorderSuite.scala [hunk 1]
```diff
@@ -537,6 +537,26 @@ trait ZorderSuite extends KyuubiSparkSQLExtensionTest with ExpressionEvalHelper
       }
     }
   }
+
+  test("Add config to control if zorder using global sort") {
```
Comment (cfmcgrady) [inline dev/kyuubi-extension-spark-common/src/test/scala/org/apache/spark/sql/ZorderSuite.scala:541]: withTable("t") {
      ...
    }

**Review by cfmcgrady — APPROVED** [review]

**Comment by ulysses-you** [comment]
thank you for review ! merged to master

**Pull request closed without merging**